test/reserve/GoodReserveCDai.test.ts

```diff
@@ -350,6 +370,93 @@ describe("GoodReserve - staking with cDAI mocks", () => {
   //   expect(gdBalanceAvatarAfter).to.be.equal(gdBalanceAvatarBefore);
   // });
 
+ 
+  
+  it("should be able to buy gd with DAI", async () => {
+    let amount = 99e8;
+    let daiAmount = ethers.utils.parseEther("100");
+    await dai["mint(uint256)"](daiAmount);
+    await dai.approve(goodReserve.address, daiAmount);
+    expect(await nameService.getAddress("DAI")).to.be.equal(dai.address)
+    let reserveToken = await marketMaker.reserveTokens(cDAI.address);
+    let reserveBalanceBefore = reserveToken.reserveSupply;
+    let supplyBefore = reserveToken.gdSupply;
+    let rrBefore = reserveToken.reserveRatio;
+    const gdBalanceBefore = await goodDollar.balanceOf(founder.address);
+    const daiBalanceBefore = await dai.balanceOf(founder.address);
+    const cDAIBalanceReserveBefore = await cDAI.balanceOf(goodReserve.address);
+    const priceBefore = await goodReserve["currentPrice()"]();
+    
+    let transaction = await (
+      await goodReserve.buy(dai.address, daiAmount, 0 , 0, NULL_ADDRESS)
+    ).wait();
+    reserveToken = await marketMaker.reserveTokens(cDAI.address);
+    let reserveBalanceAfter = reserveToken.reserveSupply;
+    let supplyAfter = reserveToken.gdSupply;
+    let rrAfter = reserveToken.reserveRatio;
+    const gdBalanceAfter = await goodDollar.balanceOf(founder.address);
+    const daiBalanceAfter = await dai.balanceOf(founder.address);
+    const cDAIBalanceReserveAfter = await cDAI.balanceOf(goodReserve.address);
+    const priceAfter = await goodReserve["currentPrice()"]();
+    expect(
+      (cDAIBalanceReserveAfter - cDAIBalanceReserveBefore).toString()
+    ).to.be.equal(amount.toString());
+    expect(
+      reserveBalanceAfter.sub(reserveBalanceBefore).toString()
+    ).to.be.equal(amount.toString());
+    expect(supplyAfter.sub(supplyBefore).toString()).to.be.equal(
+      gdBalanceAfter.sub(gdBalanceBefore).toString()
+    );
+    expect(rrAfter.toString()).to.be.equal(rrBefore.toString());
+    expect(gdBalanceAfter.gt(gdBalanceBefore)).to.be.true;
+    expect(daiBalanceBefore.gt(daiBalanceAfter)).to.be.true;
+    expect(priceAfter.toString()).to.be.equal(priceBefore.toString());
```

Comment:
all places where this repeats is wrong or requires a comment explaining why price doesnt change